### PR TITLE
Fix getPairs input parameters helper and add tests

### PR DESCRIPTION
**Pull request by dschwen**
This moves the helper around so it can be used from actions (the contact actions are a candidate, and I already have a PR ready for them), fixes the error messages, check to prevent fetching of controllable parameters, and adds unit tests for the functionality itself and the error checking.

This also adds support for MultiMooseEnums, which result in a pair component of type MooseEnumItem. This will facilitate the use of this helper fit the PETSc options iname/value parameter pair (again, I already have a PR ready for that).

Refs #18456

### CODE REVIEW

File: unit/src/InputParametersTest.C [hunk 1]
```diff
@@ -301,3 +301,68 @@ TEST(InputParameters, setPPandVofPP)
 
   // EXPECT_EQ(p1.getDefaultPostprocessorValue("pp_name"), 1);
 }
+
+TEST(InputParameters, getPairs)
```
Comment (lindsayad) [inline unit/src/InputParametersTest.C:305]: love the unit testing

File: framework/include/utils/InputParameters.h [hunk 2]
```diff
@@ -1574,6 +1589,38 @@ InputParameters::getParamHelper(const std::string & name,
   return pars.get<std::vector<T>>(name);
 }
 
+template <typename R1, typename R2, typename V1, typename V2>
+std::vector<std::pair<R1, R2>>
+InputParameters::getPairs(const std::string & param1, const std::string & param2) const
+{
+  auto v1 = get<V1>(param1);
+  auto v2 = get<V2>(param2);
```
[the remark this answers is not in the document]
Comment (dschwen) [reply framework/include/utils/InputParameters.h:1597]: you're completely right

File: unit/src/InputParametersTest.C (hunk 1 above)
Comment (dschwen) [reply unit/src/InputParametersTest.C:305]: actually, I should add a unit test for the MultiMooseEnum case...

Comment (dschwen) [reply unit/src/InputParametersTest.C:306]: done

**Comment by YaqiWang** [comment]
Let me know if you need help for patching Griffin.

**Review by lindsayad — APPROVED** [review]

**Comment by dschwen** [comment]
Thanks @YaqiWang , I'll tag you in the other PR once this is merged and I have rebased it!

**Comment by lindsayad** [comment]
@dschwen feel free to merge when tests pass and you're content with your work 😄

**Pull request merged**